python_modules/dagster/dagster/_core/definitions/run_request.py

```diff
@@ -276,6 +289,11 @@ def partition_key_range(self) -> Optional[PartitionKeyRange]:
         else:
             return None
 
+    def requires_backfill_daemon(self):
+        # TODO - can probably have a more intelligent way of determining if a run request should go to the backfill daemon
+        # maybe if there are multiple partitions keys in the asset graph subset?
+        return self.asset_graph_subset is not None
```

Comment:
for now we will keep the behavior so that all run requests with an asset graph subset get directed to the backfill daemon. we can do a follow up to update this method to introspect on the asset graph subset to determine if it can be run in a single run

Reply:
makes sense to me 👍